### PR TITLE
fix: virt_text_pos to inline or overlay

### PR DESCRIPTION
**Pull request by phgz**
I tried to simplify the code while keeping the logic.

It solves the error of #5, but I now sometimes have glitch while writing.

First screenshot: OK behaviour.
Screenshot 2023-05-25 at 00 59 46

Second screenshot: Glitch.
Screenshot 2023-05-25 at 01 01 29

**Edit**: I also noticed that `inline` produces flicker.

### CODE REVIEW

File: autoload/ddc/ui/inline.vim [hunk 1]
```diff
@@ -23,32 +23,20 @@ function! ddc#ui#inline#_show(pos, items, highlight) abort
...
-      const col = '.'->col()
-    endif
+  let head_matched = word->stridx(complete_str) == 0 && '.'->col() == '$'->col()
+  let word = has('nvim-0.10') ? remaining : word
+
+  if exists('*nvim_buf_set_extmark')
+    const col = head_matched || has('nvim-0.10') ? '.'->col() - 1 : 0
+    const options = #{
+          \   virt_text: [[word, a:highlight]],
+          \   virt_text_pos: !head_matched && has('nvim-0.10') ? 'inline' : 'overlay',
```
Comment (Shougo) [inline autoload/ddc/ui/inline.vim:33]: The line is too long.

**Comment by Shougo** [comment]
I have tested your branch.  Hm.  `inline` needs update when user input.
But it adds the flicker and it is not useful.

**Comment by Shougo** [comment]
Please remove `inline` support.
It does not work.

File: autoload/ddc/ui/inline.vim [hunk 2]
```diff
@@ -23,32 +23,20 @@ function! ddc#ui#inline#_show(pos, items, highlight) abort
...
-            \   virt_text: [[word, a:highlight]],
-            \   virt_text_pos: 'overlay',
-            \   hl_mode: has('nvim-0.10') ? 'inline' : 'combine',
-            \   priority: 0,
-            \   right_gravity: v:false,
-            \ }
-    else
-      const col = '.'->col()
-    endif
+  let head_matched = word->stridx(complete_str) == 0 && '.'->col() == '$'->col()
```
Comment (Shougo) [inline autoload/ddc/ui/inline.vim:26]:
```suggestion
  const head_matched = word->stridx(complete_str) == 0 && '.'->col() == '$'->col()
```

**Comment by Shougo** [comment]
I will improve the PR.

**Comment by phgz** [comment]
> Please remove inline support.

All good! On my side I was thinking of using `overlay` anyway as it's less intrusive.

**Comment by Shougo** [comment]
Hi.  I have improved your code in #7.  Closing.

**Comment by Shougo** [comment]
I have tested all `overlay`, but it is not good.

**Comment by phgz** [comment]
Thanks! I will give a try to `inline` virt_text_pos, I will see if the flicker doesn't bother me too much.

**Pull request closed without merging**